app/services/unwithdraw_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class UnwithdrawService
+  def process(edition, user = nil)
+    Document.transaction do
+      edition.document.lock!
+      check_publishable(edition)
```

Comment:
This should probably be check_unwithdrawable, but you can probably skip having this as a method and just have `raise "edition must be withdrawn to be unwithdrawn" unless edition.withdrawn?` on this line.